client/landing/stepper/declarative-flow/internals/steps-repository/woo-transfer/index.tsx

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable no-console */
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect } from 'react';
+import { useSite } from 'calypso/landing/stepper/hooks/use-site';
+import { transferStates } from 'calypso/state/atomic/transfers/constants';
+import { ONBOARD_STORE, SITE_STORE } from '../../../../stores';
+import type { Step } from '../../types';
+
+const wait = ( ms: number ) => new Promise( ( res ) => setTimeout( res, ms ) );
+
+const WooTransfer: Step = function WooTransfer( { navigation } ) {
+	const { submit } = navigation;
+	const { setPendingAction, setProgress } = useDispatch( ONBOARD_STORE );
+	const {
+		requestAtomicSoftwareStatus,
+		requestLatestAtomicTransfer,
+		initiateAtomicTransfer,
+	} = useDispatch( SITE_STORE );
+	const site = useSite();
+
+	const siteId = site?.ID;
+
+	const {
+		getSiteLatestAtomicTransfer,
+		getSiteLatestAtomicTransferError,
+		getAtomicSoftwareStatus,
+		// getAtomicSoftwareError,
+	} = useSelect( ( select ) => select( SITE_STORE ) );
+
+	useEffect( () => {
+		if ( ! siteId ) return;
+
+		setPendingAction( async () => {
+			const startTime = new Date().getTime();
+			const totalTimeout = 1000 * 180;
+			const maxFinishTime = startTime + totalTimeout;
+
+			// Initiate transfer
+			await initiateAtomicTransfer( siteId, 'woo-on-plans' );
```

Comment:
why is `await` needed here?

Reply:
It's not _needed_, the only reason is to make sure the process was initiated before start polling for the result.

Reply:
I'll try to figure that out!
In any case, do you think the `await` there is harmful in any way?

Reply:
I don't really know. But if it isn't needed I think it's best to be removed.

Reply:
I'm not super familiar with generators. With that said, event hough we implement actions as generators, I don't think we access the generators directly in the client code (using `useDispatch`). Considerations:

- we don't have access to the generator object (that we can call `next` and get to the next `yield`).
- TS identifies, for ex, `initiateAtomicTransfer` as a regular function that returns a `Promise<void>`

My guess is that `@wordpress/data` manages the generators and calls `.next()` until `done == true` under the hood, returning a `Promise` after that.

Reply:
I did some testing and `await`ing for the action calls works. With `await` it's gonna wait until everything is `yield`ed (requests completed, etc) before continuing. the same doesn't happen without `await`.